cmd/admin-handlers.go

```diff
@@ -1779,3 +1786,89 @@ func (a adminAPIHandlers) TraceHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+
+	vars := mux.Vars(r)
+	switch infoType := vars["serviceType"]; infoType {
+	case "vault":
+		errorString := ""
+		if GlobalKMS == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrKMSNotConfigured), r.URL)
+			return
+		}
+		config, err := readServerConfig(ctx, objectAPI)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+		endPoint := Environment.Get(EnvVaultEndpoint, config.KMS.Vault.Endpoint)
+		vaultName := Environment.Get(EnvVaultKeyName, config.KMS.Vault.Key.Name)
+		vaultType := Environment.Get(EnvVaultAuthType, config.KMS.Vault.Auth.Type)
+
+		// check for online/offline status of endpoint
+		if err := checkStatus(endPoint); err != nil {
+			errorString = err.Error()
+		}
+
```

Comment:
Re-reading config is not required in this case. `endpoint`, `vaultName` and `vaultType` can be directly read from `GlobalKMS` like,
```go
endpoint := GlobalKMS.Vault.Endpoint
vaultName := GlobalKMS.Vault.Key.Name
vaultType := GlobalKMS.Vault.Auth.Type
```